### DISCUSSION TITLE
Ping Pong mechanism

### DISCUSSION THREAD

**Discussion by chuwt** [socketio/socket.io#4056] [Q&A] [answered]
I wanner know how server send ping packet when using HTTP long-polling transport, is there a doc to show the detail?

**Comment by darrachequesne** [comment, marked as answer]
Hi! The documentation could indeed be more detailed on this point.

Here's what happens:

- the server sends a ping packet (a "2") on the ongoing GET request
- the client sends a pong packet back (a "3") with a POST request
- the client then opens another long-running GET request

You can take a look at the schema here: https://socket.io/docs/v4/how-it-works/#HTTP-long-polling

**Reply by chuwt** [reply]
Thanks for your reply! Do you mean:
1. client send `Get` request to server
2. server received and send `ping packet` to client
3. client received `ping packet` and send `pong packet` request with `Post`

**Reply by darrachequesne** [reply]
Yes that's right :+1: 

The GET request is a long-running request, that is left open until the server wants to send some data to the client, hence the "long-polling" name.

**Reply by chuwt** [reply]
Thanks for your patience, it's clear for me now.

**Reply by awesomelike** [reply]
@darrachequesne you said "GET request is a long-running request", how long it can be? Something like more than 60 seconds? I have a question about this here: https://github.com/socketio/socket.io/discussions/4076
